### PR TITLE
file upload edit functionality added

### PR DESCRIPTION
**Pull request by vinceli1004**
Allowed users to upload files to edit fields as opposed to manually typing.

### CODE REVIEW

File: frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts [hunk 1]
```diff
@@ -401,9 +401,207 @@ export class ChallengesettingsComponent implements OnInit, OnDestroy {
...
+      FORM_DATA.append('overview_file', params['overview_file']);
+      SELF.apiService
+        .patchFileUrl(
+          SELF.endpointsService.editChallengeDetailsURL(SELF.challenge.creator.id, SELF.challenge.id),
+          FORM_DATA
+        )
+        .subscribe(
+          (data) => {
+            SELF.challenge.description = data.description;
+            SELF.globalService.showToast('success', 'The description is successfully updated!', 5);
```
Comment (Ram81) [inline frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts:419]: Please change the message to `Challenge description updated successfully!`

File: frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts [hunk 2]
```diff
@@ -401,9 +401,207 @@ export class ChallengesettingsComponent implements OnInit, OnDestroy {
...
+      FORM_DATA.append('terms_and_conditions_file', params['terms_and_conditions_file']);
+      SELF.apiService
+        .patchFileUrl(
+          SELF.endpointsService.editChallengeDetailsURL(SELF.challenge.creator.id, SELF.challenge.id),
+          FORM_DATA
+        )
+        .subscribe(
+          (data) => {
+            SELF.challenge.terms_and_conditions = data.terms_and_conditions;
+            SELF.globalService.showToast('success', 'The terms and conditions are successfully updated!', 5);
```
Comment (Ram81) [inline frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts:468]: Please change the message to `Terms and conditions updated successfully!`

File: frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts [hunk 3]
```diff
@@ -401,9 +401,207 @@ export class ChallengesettingsComponent implements OnInit, OnDestroy {
...
+          },
+          () => this.logger.info('EDIT-TERMS-AND-CONDITIONS-FINISHED')
+        );
+    };
+    
+    /**
+     * Parameters of the modal
+     */
+    const PARAMS = {
+      title: 'Edit Challenge Terms and Descriptions',
```
Comment (Ram81) [inline frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts:482]: The title should be `Edit Terms and Conditions`

File: frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts [hunk 4]
```diff
@@ -401,9 +401,207 @@ export class ChallengesettingsComponent implements OnInit, OnDestroy {
...
+      FORM_DATA.append('evaluation_criteria_file', params['evaluation_criteria_file']);
+      SELF.apiService
+        .patchFileUrl(
+          SELF.endpointsService.editChallengeDetailsURL(SELF.challenge.creator.id, SELF.challenge.id),
+          FORM_DATA
+        )
+        .subscribe(
+          (data) => {
+            SELF.challenge.evaluation_details = data.evaluation_details;
+            SELF.globalService.showToast('success', 'The evaluation details is successfully updated!', 5);
```
Comment (Ram81) [inline frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts:517]: Please change the message to `Evaluation details updated successfully!`

File: frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts [hunk 5]
```diff
@@ -401,9 +401,207 @@ export class ChallengesettingsComponent implements OnInit, OnDestroy {
...
+          },
+          () => this.logger.info('EDIT-CHALLENGE-EVALUATION-DETAILS-FINISHED')
+        );
+    };
+
+    /**
+     * Parameters of the modal
+     */
+    const PARAMS = {
+      title: 'Edit Challenge Evaluation Criteria',
```
Comment (Ram81) [inline frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts:531]: Change the title to `Edit Evaluation Criteria`

File: frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts [hunk 6]
```diff
@@ -401,9 +401,207 @@ export class ChallengesettingsComponent implements OnInit, OnDestroy {
...
+      content: '',
+      confirm: 'Submit',
+      deny: 'Cancel',
+      form: [
+        {
+          name: 'Challenge Evaluation Criteria',
```
Comment (Ram81) [inline frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts:537]: same here

File: frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts [hunk 7]
```diff
@@ -401,9 +401,207 @@ export class ChallengesettingsComponent implements OnInit, OnDestroy {
...
+        .patchFileUrl(
+          SELF.endpointsService.updateChallengePhaseDetailsURL(SELF.challenge.id, SELF.selectedPhase['id']),
+          FORM_DATA
+        )
+        .subscribe(
+          (data) => {
+            for (const attrname of Object.keys(data)) {
+              SELF.selectedPhase[attrname] = data[attrname];
+            }
+            SELF.globalService.showToast('success', 'The challenge phase description is successfully updated!');
```
Comment (Ram81) [inline frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts:569]: Please change the message to `Challenge phase description updated successfully!`

**Review by Ram81 — CHANGES_REQUESTED** [review]
Some minor comments

File: frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts (hunk 1 above)
Comment (vinceli1004) [reply frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts:419]: Addressed.

**Comment by RishabhJain2018** [comment]
@Ram81 @vinceli1004 is this PR ready for merge?

**Review by Ram81 — APPROVED** [review]
LGTM

**Review by RishabhJain2018 — APPROVED** [review]

**Pull request merged**